src/Microsoft.DotNet.Darc/src/DarcLib/AzureDevOpsClient.cs

```diff
@@ -759,5 +763,153 @@ public Task CommitFilesAsync(List<GitFile> filesToCommit, string repoUri, string
+        /// <param name="build">Build which should be added as source of the release definition.</param>
+        /// <returns>AzureDevOpsReleaseDefinition</returns>
+        public async Task<AzureDevOpsReleaseDefinition> AddArtifactSourceAsync(string accountName, string projectName, AzureDevOpsReleaseDefinition releaseDefinition, AzureDevOpsBuild build)
+        {
+            releaseDefinition.Artifacts = new AzureDevOpsArtifact[1] {
+                new AzureDevOpsArtifact()
+                {
+                    Alias = "PrimaryArtifact",
+                    Type = "Build",
+                    DefinitionReference = new AzureDevOpsArtifactSourceReference()
+                    {
+                        Definition = new AzureDevOpsIdNamePair()
+                        {
+                            Id = build.Definition.Id.ToString(),
+                            Name = build.Definition.Name
+                        },
+                        DefaultVersionType = new AzureDevOpsIdNamePair()
+                        {
+                            Id = "specificVersionType",
+                            Name = "Specific version"
+                        },
+                        DefaultVersionSpecific = new AzureDevOpsIdNamePair()
+                        {
+                            Id = build.Id.ToString(),
+                            Name = build.BuildNumber
+                        },
+                        Project = new AzureDevOpsIdNamePair()
+                        {
+                            Id = build.Project.Id.ToString(),
+                            Name = build.Project.Name
+                        }
+                    }
+                }
+            };
+
+            var _serializerSettings = new JsonSerializerSettings
+            {
+                ContractResolver = new CamelCasePropertyNamesContractResolver(),
+                NullValueHandling = NullValueHandling.Ignore
+            };
+
+            var body = JsonConvert.SerializeObject(releaseDefinition, _serializerSettings);
+
+            JObject content = await this.ExecuteRemoteGitCommandAsync(
+                HttpMethod.Put,
+                accountName,
+                projectName,
+                $"_apis/release/definitions/",
+                _logger,
+                body,
+                versionOverride: "5.0-preview.3",
+                baseAddressSubpath: "vsrm.");
+
+            return content.ToObject<AzureDevOpsReleaseDefinition>();
+        }
+
+        /// <summary>
+        ///     Remove all artifact sources of the release definition informed.
+        /// </summary>
+        /// <param name="accountName">Azure DevOps account name</param>
+        /// <param name="projectName">Project name</param>
+        /// <param name="releaseDefinition">Release definition to be modified</param>
+        public async void RemoveAllArtifactSourcesAsync(string accountName, string projectName, AzureDevOpsReleaseDefinition releaseDefinition)
```

Comment:
Are there worries about concurrency here?  What happens if the release pipeline gets launched by two builds in rapid succession?

Reply:
I had a prototype running locally and I was able to enqueue several releases in succession for different builds and I didn't get any problem.